### DISCUSSION TITLE
Completely restoring grid state from local or session storage

### DISCUSSION THREAD

**Discussion by jkixj** [vikramlearning/blazorbootstrap#477] [Q&A]
Hi, I have a grid that is enabled for sorting, filtering, and pagination. I am using a DataProvider to fetch the data.

I am able to restore the filters and pagination (GridSettings) on the grid by loading those from session settings, but I need to be able to also restore sorting as well. The idea being that the user can return the a screen and it (seem to) be in the same state as they left it.

My understanding is that sorting is a property of each column, but there is no way to dynamically access the columns given a reference to the grid.

Is there a way to do this other than manually tracking settings for each column? This seems like something that should be part of the GridSettings API, but it is not.

**Comment by gvreddy04** [comment]
@jkixj Thank you for using BlazorBootstrap. Saving the Grid's sorting state as part of GridSettings is not yet supported. This functionality will be included in future releases.

Link: https://docs.blazorbootstrap.com/components/grid#save-and-load-grid-settings

![image](https://github.com/vikramlearning/blazorbootstrap/assets/2337067/dc196c54-cc9b-4a7d-9dd7-7fcd92c81613)

**Reply by jkixj** [reply]
Thank you. I actually had read that warning, but forgot to include that in the question. I guess I was looking for a workaround until then.

**Comment by gvreddy04** [comment]
@jkixj Please raise an issue for this. We will implement an enhancement to address it.

**Reply by jkixj** [reply]
Thank you, I will work on that.
